amr-wind/immersed_boundary/bluff_body/ghost_cell.H

```diff
@@ -67,37 +67,30 @@ AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void vandermonde_matrix(
  * Here, A is the coefficient matrix, and x is the unknown.
  *
  */
-
+template <class T, int Nm1>
 AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void gauss_siedel_method(
-    amrex::Array2D<amrex::Real, 0, 7, 0, 7>& A,
+    const amrex::Array2D<T, 0, Nm1, 0, Nm1>& A,
```

Comment:
@gdeskos : The function is now templated such that it can be used to solve linear system of arbitrary size N.

Reply:
Thanks, this is only going to take reals right ? I ll keep it like that for the moment for we shouldn't need the templating. We might move it to core if we need to share it with other classes as well

Reply:
@gdeskos Thanks. Yes, the function takes in only real inputs.